numpy/core/tests/test_regression.py

```diff
@@ -452,6 +452,10 @@ def test_lexsort_zerolen_custom_strides_2d(self):
         xs.strides = (16, 16)
         assert np.lexsort((xs,), axis=0).shape[0] == 2
 
+    def test_lexsort_invalid_axis(self):
+        assert_raises(np.AxisError, np.lexsort,(np.arange(1),), axis=2)
+        assert_raises(np.AxisError, np.lexsort,(np.array([]),), axis=1)
```

Comment:
NIT: Missing space. We may want to test the weird 0-D behaviour as well. I am willing to consider it a bug, but maybe we should deprecate it anyway.
Just a side note: `assert_raises` can be used as a context manager if you prefer, doubt it matters here.